src/target/ext/base/multithread.rs

```diff
@@ -290,3 +313,25 @@ pub trait MultiThreadRangeStepping: Target + MultiThreadResume {
 }
 
 define_ext!(MultiThreadRangeSteppingOps, MultiThreadRangeStepping);
+
+/// Target Extension - support for GDB's "Scheduler Locking" mode.
+/// See [`MultiThreadResume::support_scheduler_locking`].
+pub trait MultiThreadSchedulerLocking: Target + MultiThreadResume {
+    /// Configure the target to enable "Scheduler Locking" for the upcoming
+    /// resume.
+    ///
+    /// This method is called when the GDB client sends a `vCont` packet that
```

Comment:
In general, `gdbstub`'s docs try to avoid leaking the nitty-gritty of the underlying protocol to the end user. Interested users can always crack the code open to figure out these details... but these docs are expected to answer high-level "what is this, how is this useful" questions.

Instead - I would remove this paragraph, and maybe include some phrasing that mentions that this behavior is toggled on/off in GDB by using the `set scheduler-locking {on/off}` command.